apps/ltn/src/per_resident_impact.rs

```diff
@@ -0,0 +1,395 @@
+                    .text_widget(ctx)
+                    .centered_vert(),
+                ColorLegend::gradient(
+                    ctx,
+                    &scale,
+                    vec!["0", &max_change.to_string(&UnitFmt::metric())],
+                )
+                .centered_vert(),
+                ColorLegend::row(ctx, *colors::PLAN_ROUTE_BEFORE, "before changes"),
+                ColorLegend::row(ctx, *colors::PLAN_ROUTE_AFTER, "after changes"),
+            ]);
+        }
+        self.bottom_panel = BottomPanel::new(ctx, &self.appwide_panel, Widget::row(row));
+
+        let map = &app.per_map.map;
+        self.world = World::bounded(map.get_bounds());
+
+        for b in map.all_buildings() {
+            if let Some((before, after)) = self.times_from_building.get(&b.id) {
+                let color = scale.eval((*after - *before) / max_change);
+                let mut txt = Text::from(if before == after {
+                    format!("No change -- {before}")
+                } else {
+                    format!(
+                        "{} slower -- {before} before this proposal, {after} after",
+                        *after - *before
+                    )
+                });
+                if before != after {
+                    txt.add_line(Line("Click").fg(ctx.style().text_hotkey_color));
+                    txt.append(Line(" to investigate"));
+                }
+
+                self.world
+                    .add(Obj::Building(b.id))
+                    .hitbox(b.polygon.clone())
+                    .draw_color(color)
+                    .hover_color(colors::HOVER)
+                    .tooltip(txt)
+                    .clickable()
+                    .build(ctx);
+            } else {
+                // TODO We run out of GPU memory on large maps if we hover here. Make World be
+                // lazier!
+                self.world
+                    .add(Obj::Building(b.id))
+                    .hitbox(b.polygon.clone())
+                    .drawn_in_master_batch()
+                    .invisibly_hoverable()
+                    //.hover_color(colors::HOVER)
+                    .clickable()
+                    .build(ctx);
+            }
+        }
+        self.world.initialize_hover(ctx);
+
+        if let Some(b) = self.current_target {
+            self.world.draw_master_batch(
+                ctx,
+                GeomBatch::load_svg(ctx, "system/assets/tools/star.svg")
+                    .centered_on(map.get_b(b).polygon.center()),
+            );
+        }
+    }
+
+    fn recalculate_times(&mut self, app: &App, timer: &mut Timer) {
+        self.times_from_building.clear();
+        self.compare_routes = None;
+        let target = if let Some(b) = self.current_target {
+            b
+        } else {
+            return;
+        };
+
+        let map = &app.per_map.map;
+
+        // Depending on the number of buildings_inside, Dijkstra may be faster, but this seems fast
```

Comment:
Actually these pathfinders can be per-State and not recalculated everytime, I'll make a fix momentarily